src/flask/routes/oauth2.py

```diff
@@ -0,0 +1,47 @@
+from flask import Blueprint, request, make_response
+
+import jwt
+
+from src.constants.http_status_code import HttpStatus
+from src.env_config import get_env_config_instance
+from src.lib.repositories.impl_v2.oauth2_repository_impl import Oauth2RepositoryImpl
+from src.utils.oauth2_util import validate_roles_and_scopes
+
+
+def setup_oauth2_routes(ioc):
+
+    oauth2_blueprint = Blueprint("oauth2_blueprint", __name__)
+    oauth2_repository = ioc.get(Oauth2RepositoryImpl)
+
+    @oauth2_blueprint.before_app_request
+    def before_request_middleware():
+        endpoint = request.endpoint
+        env_config = get_env_config_instance()
+
+        try:
+            access_token = request.args.get("access_token")
+            oauth2_repository.validate_token(access_token)
+
+        except jwt.exceptions.ExpiredSignatureError:
+            response = make_response("Token expired", HttpStatus.UNAUTHORIZED.value)
+            return response
+        except jwt.exceptions.InvalidSignatureError:
+            response = make_response("Token Invalid", HttpStatus.UNAUTHORIZED.value)
+            return response
+        except jwt.exceptions.DecodeError:
+            response = make_response(
+                "Not enough Segments", HttpStatus.UNAUTHORIZED.value
+            )
+            return response
+
+        validation = validate_roles_and_scopes(
```

Comment:
Move this inside the `try/catch` since this should throw and error

Reply:
done.